samples/nrf9160/modem_shell/src/gnss/gnss_interface.c

```diff
@@ -600,50 +600,44 @@ static void handle_timeout_work_fn(struct k_work *item)
 
 static void gnss_api_init(void)
 {
-	int err;
 	static bool gnss_api_initialized;
 
 	if (gnss_api_initialized) {
+		/* Reset event handler in case some other handler was set in the meantime */
+		nrf_modem_gnss_event_handler_set(gnss_event_handler);
 		return;
 	}
 
 	/* Activate GNSS API v2 */
-	err = nrf_modem_gnss_init();
-
-	if (!err) {
-		err = nrf_modem_gnss_event_handler_set(gnss_event_handler);
-	}
+	nrf_modem_gnss_init();
```

Comment:
We are not checking the return value here for failures. I guess only failure would be if it's already initialized?
In that case it would be good to document that we are intentionally not checking it and this would do that:
(void)nrf_modem_gnss_init();

Reply:
Swell!

Reply:
`nrf_modem_gnss_event_handler_set()` can return an error as well (if GNSS is running), so maybe that should be `void` cast as well.

Reply:
Done. I also rebased the branch.